src/buffer.cc

```diff
@@ -111,6 +150,58 @@ Result Buffer::IsEqual(Buffer* buffer) const {
   return {};
 }
 
+std::vector<float> Buffer::CalculateDiffs(const Buffer* buffer) const {
+  std::vector<float> diffs;
+
+  auto* buf_1_ptr = GetValues<uint8_t>();
+  auto* buf_2_ptr = buffer->GetValues<uint8_t>();
+  auto comps = format_->GetComponents();
+
+  for (size_t i = 0; i < ElementCount(); ++i) {
+    for (size_t j = 0; j < format_->ColumnCount(); ++j) {
+      auto* buf_1_row_ptr = buf_1_ptr;
+      auto* buf_2_row_ptr = buf_2_ptr;
+      for (size_t k = 0; k < format_->RowCount(); ++k) {
+        diffs.push_back(CalculateDiff(comps[k], buf_1_row_ptr, buf_2_row_ptr));
+
+        buf_1_row_ptr += comps[k].SizeInBytes();
+        buf_2_row_ptr += comps[k].SizeInBytes();
+      }
+      buf_1_ptr += format_->SizeInBytesPerRow();
+      buf_2_ptr += format_->SizeInBytesPerRow();
+    }
+  }
+
+  return diffs;
+}
+
+Result Buffer::CompareRMSE(Buffer* buffer, float tolerance) const {
+  if (!buffer->format_->Equal(format_.get()))
+    return Result{"Buffers have a different format"};
+  if (buffer->element_count_ != element_count_)
+    return Result{"Buffers have a different size"};
+  if (buffer->width_ != width_)
+    return Result{"Buffers have a different width"};
+  if (buffer->height_ != height_)
+    return Result{"Buffers have a different height"};
+  if (buffer->ValueCount() != ValueCount())
+    return Result{"Buffers have a different number of values"};
+
+  auto diffs = CalculateDiffs(buffer);
+  float sum = 0.0;
```

Comment:
I'd do this in double.
Sums of squares can get pretty big.

We _could_ go even further and normalize.:   E.g. divide  each diff by the _mean_ of the diffs, and then compute RMSE, then remultiply by the absolute value of the mean.
But that's probably excessive for now.